automation/src/test/scala/org/broadinstitute/dsde/test/api/SnapshotAPISpec.scala

```diff
@@ -0,0 +1,143 @@
+package org.broadinstitute.dsde.test.api
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.{StatusCodes, Uri}
+import akka.http.scaladsl.model.Uri.Path
+import bio.terra.datarepo.api.RepositoryApi
+import bio.terra.datarepo.client.ApiClient
+import bio.terra.workspace.model.{DataReferenceList, ReferenceTypeEnum}
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.auth.AuthToken
+import org.broadinstitute.dsde.workbench.config.ServiceTestConfig.FireCloud
+import org.broadinstitute.dsde.workbench.config.UserPool
+import org.broadinstitute.dsde.workbench.fixture.{BillingFixtures, WorkspaceFixtures}
+import org.broadinstitute.dsde.workbench.service.Rawls
+import org.broadinstitute.dsde.workbench.service.util.Tags
+import org.scalatest.freespec.AnyFreeSpecLike
+import org.scalatest.matchers.should.Matchers
+
+import java.util.UUID
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+class SnapshotAPISpec extends AnyFreeSpecLike with Matchers
+  with WorkspaceFixtures with BillingFixtures
+  with SprayJsonSupport with LazyLogging {
+
+  private val dataRepoBaseUrl = FireCloud.dataRepoApiUrl
+
+  // assume we have a valid dataRepoBaseUrl. We've had bugs/problems in our config so this is a safeguard.
+  assume(Try(Uri(dataRepoBaseUrl)).isSuccess,
+    s"Canceling test because [$dataRepoBaseUrl] is not a valid url for data repo")
+
+  "TDR Snapshot integration" - {
+
+    "should be able to contact Data Repo" taggedAs(Tags.AlphaTest, Tags.ExcludeInFiab) in {
+      // status API is unauthenticated, but all our utility methods expect a token.
+      // so, we'll send a token to the unauthenticated API to make this code path easier.
+      val owner = UserPool.userConfig.Owners.getUserCredential("hermione")
+      implicit val ownerAuthToken: AuthToken = owner.makeAuthToken()
+      val statusRequest = Rawls.getRequest(dataRepoBaseUrl + "status")
+
+      withClue(s"Data Repo status API returned ${statusRequest.status.intValue()} ${statusRequest.status.reason()}!") {
+        statusRequest.status shouldBe StatusCodes.OK // compare
+      }
+    }
+
+    "should allow snapshot references to be added to workspaces" taggedAs(Tags.AlphaTest, Tags.ExcludeInFiab) in {
+      // as of this writing, hermione.owner is the user with access to snapshots
+      val owner = UserPool.userConfig.Owners.getUserCredential("hermione")
+
+      implicit val ownerAuthToken: AuthToken = owner.makeAuthToken()
+
+      withCleanBillingProject(owner) { projectName =>
+        withWorkspace(projectName, s"${UUID.randomUUID().toString}-snapshot references") { workspaceName =>
+
+          // call data repo to list snapshots
+          // this gets the most recent snapshots in TDR (to which we have read access). This can cause tests to change
```

Comment:
Curious if using the most recent snapshots is considered a preferred approach...versus using dedicated snapshots?
If this test (or future tests using this approach) need to assert on snapshot details for example, then the result of that assertion may not always be predictable i.e. we won't know the state of the most recent snapshots and if they are suitable for particular tests intending to use them.

I understand that using dedicated snapshots may also have its drawbacks.

Reply:
I'm trying very hard to not assume anything about the data that exists in TDR - I'd rather everything be dynamic. If/when we  need to assert on snapshot details, I'd implement it by making a request directly to TDR for the details, then a request to Rawls for details, and compare the two responses.

Reply:
That's fair.